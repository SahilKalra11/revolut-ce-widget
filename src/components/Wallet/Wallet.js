import React, {Component} from 'react';
import './Wallet.scss';
import {getCurrencySymbol} from "../../helpers";

class Wallet extends Component {
    render() {
        const {wallet, currencyList} = this.props;
        return (
            <section className="wallet">
                <h3>Your wallet:</h3>
                <ul className="wallet__list">
                    {currencyList.map((currencyItem, index) => {
                        const key = `${currencyItem}${index}`;
                        const value = `${getCurrencySymbol(currencyItem)}${wallet[currencyItem]}`;
                        return (
                            <li className="wallet__item" key={key}><b>{currencyItem}</b>:{value}</li>
                        )
                    })
                    }
                </ul>
            </section>
        );
    }
}

export default Wallet;
