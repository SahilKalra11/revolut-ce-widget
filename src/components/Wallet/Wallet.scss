@import '../../utilities/variables';

.wallet {
  background: $color-white;
  color: $color-blue;
  padding: $padding-sm $padding;
  border-radius: 40px;
  li{
    font-size: 15px;
    font-weight: 600;
  }
  h3{
    font-size: 1rem;
  }
  &__list {
    display: flex;
    margin: 0;
  }

  &__item {
    list-style: none;
    margin-right: $margin;
    margin:auto
  }
}

@media only screen and (min-width: 414px){
  .wallet {
    max-width: 400px;
  width: 100%;
  margin: auto;
  }
}