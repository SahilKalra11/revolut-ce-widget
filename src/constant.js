export const AVAILABLE_CURRENCY_LIST = ['USD', 'GBP', 'EUR'];
export const APP_ID = '115aba0dc4cb4ceaa4c38c5104e3cce7';
export const BASE_URL = 'https://openexchangerates.org/api/latest.json?app_id=';
export const TEST_RATES = {USD: 1, EUR: 1.20, GBP: 1.3};
export const TEST_WALLET = {USD: 100, EUR: 100, GBP: 100,};

