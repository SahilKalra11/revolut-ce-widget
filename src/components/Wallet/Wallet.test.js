import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import Wallet from './Wallet';
import {unmountComponentAtNode} from "react-dom";
import {AVAILABLE_CURRENCY_LIST} from "../../constant";

describe('ExchangeInput', () => {
    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    const props = {
        wallet: {
            USD: 100,
            EUR: 100,
            GBP: 100,
        },
        currencyList: AVAILABLE_CURRENCY_LIST,
    };

    it('renders component without crashing', () => {
        ReactDOM.render(
            <Wallet {...props} />,
            container
        );
    });

    test('matches to snapshot', () => {
        const component = render.create(
            <Wallet {...props} />
        );
        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });
});
