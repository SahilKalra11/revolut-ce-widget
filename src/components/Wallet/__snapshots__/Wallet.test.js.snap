// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeInput matches to snapshot 1`] = `
<section
  className="wallet"
>
  <h3>
    Your wallet:
  </h3>
  <ul
    className="wallet__list"
  >
    <li
      className="wallet__item"
    >
      <b>
        USD
      </b>
      :
      $100
    </li>
    <li
      className="wallet__item"
    >
      <b>
        GBP
      </b>
      :
      £100
    </li>
    <li
      className="wallet__item"
    >
      <b>
        EUR
      </b>
      :
      €100
    </li>
  </ul>
</section>
`;
